### PR TITLE
dot, dot/core: migrate core types to dot types

### LINKED ISSUE #722: investigate and clean up core types
## Expected Behavior

If we decide to keep the `types` package within `core`, it should include types specific to `core`.

All `types` packages within a package should be specific to that package and should not be used throughout the code as the `types` package within `core` is currently being used.

## Current Behavior

The `types` package within `core` is currently being used throughout the codebase and its types should either be migrated to `common` types or `dot` types (seems to be more appropriate than within `core` if not `common`).

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes
- migrate core types to dot types and organize types into individual files

## Tests:
```
go test ./... -short
```

### Issues:
closes #722

### CODE REVIEW

File: dot/types/bytes_test.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
```
Comment (noot) [inline dot/types/bytes_test.go:1]: remove?

File: dot/types/extrinsics.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
```
Comment (noot) [inline dot/types/extrinsics.go:1]: name extrinsic.go

File: go.mod [hunk 3]
```diff
@@ -6,20 +6,21 @@ require (
 	github.com/ChainSafe/go-schnorrkel v0.0.0-20200115165343-aa45d48b5ed6
 	github.com/ChainSafe/log15 v1.0.0
 	github.com/OneOfOne/xxhash v1.2.5
+	github.com/allegro/bigcache v1.2.1 // indirect
```
Comment (noot) [inline go.mod:9]: any reason the go mods changed?

File: lib/common/address.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package common
+
+// Address represents a base58 encoded public key
+type Address string
```
Comment (noot) [inline lib/common/address.go:20]: not sure how necessary this is, but ok

**Comment by ryanchristo** [comment]
cc @thomasmodeneis and/or @edwardmack this pull request is ready for review when you have a moment. I will address comments from @noot. :pray:

File: go.mod (hunk 3 above)
Comment (ryanchristo) [reply go.mod:9]: I ran go mod tidy at some point. I can revert if necessary.

**Pull request merged**